src/history/hs_rec.c

```diff
@@ -645,7 +645,7 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
              * the RTS.
              */
             nentries = MAX_REVERSE_MODIFY_NUM;
-            if (!F_ISSET(upd, WT_UPDATE_DS) && !F_ISSET(prev_upd, WT_UPDATE_DS) &&
+            if (!F_ISSET(upd, WT_UPDATE_DS) && prev_upd != list->onpage_upd &&
```

Comment:
I think this still not considering the squash case. We should compare the txnid and timestamp with the onpage_upd.

Reply:
Can this be a separate ticket or should we fix it in this ticket?

Reply:
We have agreed to make a separate ticket for this.